ext/bit/bit_details.hpp

```diff
@@ -754,6 +763,78 @@ constexpr T _bitblend(T src0, T src1, T start, T len) noexcept
 
 
 
+// ---------- IMPLEMENTATION DETAILS: INSTRUCTIONS: BIT EXCHANGE ------------ //
+// Exchanges/swaps bits of src0 by the ones of src1 where the mask is true
+template <class T>
+constexpr void _bitexch(T& src0, T& src1, T msk) noexcept
+{ 
+    src0 = src0 ^ static_cast<T>(src1 & msk);
```

Comment:
I'm fairly certain we don't need (any?) of the `static_cast` calls but it's not really worth holding up a merge over

Reply:
I think you're right. It was probably only an issue when I was using bitwise operations and then calling functions on the intermediates before they were casted back. I don't mind going back and getting rid of them but I'll do it in the next PR.